tensorflow_addons/layers/stochastic_depth.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+from typeguard import typechecked
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class StochasticDepth(tf.keras.layers.Layer):
+    r"""Stochastic Depth layer.
+
+    Implements Stochastic Depth as described in
+    [Deep Networks with Stochastic Depth](https://arxiv.org/abs/1603.09382), to randomly drop residual branches
+    in residual architectures.
+
+    Usage:
+    Residual architectures with fixed depth, use residual branches that are merged back into the main network
+    by adding the residual branch back to the input:
+
+    ```python
+    residual = tf.keras.layers.Conv2D(...)(input)
+
+    return tf.keras.layers.Add()([input, residual])
+    ```
+
+    StochasticDepth acts as a drop-in replacement for the addition:
+
+    ```python
+    residual = tf.keras.layers.Conv2D(...)(input)
+
+    return tfa.layers.StochasticDepth()([input, residual])
+    ```
```

Comment:
For python code snippet, we encourage to use testable docstring with doctest. See https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#testing-docstrings .